doc/ref_arch/kubernetes/chapters/appendix-a.md

```diff
@@ -1,18 +1,26 @@
 [<< Back](../../kubernetes)
-# Appendix A - Guidance for VNF-to-CNF Transition
+# Appendix A - Guidelines For VNF Vendors
 <p align="right"><img src="../figures/bogo_ifo.png" alt="scope" title="Scope" width="35%"/></p>
 
 ## Table of Contents
-* [A.1 TBC](#A.1)
-* [A.2 TBC](#A.2)
+
+* [A.1 Exception List](#A.1)
+* [A.2 Guidance for VNF-to-CNF Transition](#A.2)
 * [A.3 TBC](#A.3)
 * [A.4 TBC](#A.3)
 
 <a name="A.1"></a>
-## A.1 TBC
+## A.1 Exception List
+
+| Ref         | Type       | Name               | RM Exception | Details |
+|-------------|------------|--------------------|--------------|---------|
+| ra2.exc.001 | Technology | SR-IOV             | rm.exc.001   | This exception allows workloads (VNF/CNF) to use SR-IOV. RA2 architectures allows plugins such as SR-IOV CNI and SR-IOV related Device Plugins, like [sriov-network-device-plugin](https://github.com/intel/sriov-network-device-plugin) to be used. |
```

Comment:
Hi @CsatariGergely,
A) Can you clarify why this exception required? What type of kernel driver bindings do you envision? I'm asking because I didn't see any configurations of the sriov-network-device-plugin that violated any CNTT Principles, and you explicitly removed references to PCI-PT (which would require an exception), so you may simply be able to permanently codify your design in RA-2, without the need to draw out and maintain a formal exception. There's a related discussion in PR #1336 

B) If we're creating a new Exception, the intent is to record the rationale and need for the exception. Also, the exception should identify when the exception will be retired, and what the transition plan looks like. I'd suggest you use the draft template in the exception framework in the RM, as you can fill in the blanks and won't miss anything.

C) If you're replicating an existing exception, can it be incorporated by reference, so the 2 copies don't have to be kept in sync?

Thanks,
-Mark

Reply:
A) sriov-network-device-plugin assigns PCI devices to the containers and the container needs to understand how to consume them. I've removed the PCI-PT references to address the comments from @rabi-abdel and @walterkozlowski and added the sriov-network-device-plugin to "soften" the PCI-PT a bit (bit in reality they are kinda the same). 
B) Okey, if we need a new exception I will do that. 
C) The original intent was to just copy the RM exception, but I agree that a) we should make the RM exceptions linkable and b) only refer to them in the RA exception list, or just do not have an RA exception list. 

For me the best solution would be to merge #1354 and just refer it here or remove the RA2 exception list completely.

@walterkozlowski , @rabi-abdel ?

Reply:
@CsatariGergely Hi Gergely, 
>
> A) sriov-network-device-plugin assigns PCI devices to the containers and the container needs to understand how to consume them. 

Agreed, containers do need to understand how to consume PCI-coupled resources, which describes many resources in a server. However, as you know better than I, host-based kernel drivers expose them as an abstraction, so they are normalized by the time the container deals with them. Based on the sriov-network-device-plugin docs, the actual h/w drivers are instantiated in the host, where they are bound to the actual PCI devices. Here's an excerpt of the workflow from their docs, [here](https://github.com/intel/sriov-network-device-plugin):

- Load device's (Physical function if it is SRIOV capable) kernel module and bind the driver to the PF
- Create required Virtual functions
- Bind all VF with right drivers

I interpret the context of those items as being in the host. So, if their docs are accurate, and I'm reading them correctly, sriov-network-device-plugin gives us more value than a simple PCI-PT transport.

>
>  "soften" the PCI-PT a bit (bit in reality they are kinda the same).

Again, based on what I see in their docs, the only thing I see the two have in common is they allocate (dedicate) a VF (PCI device) to a process. We do that all over the place, notably in virtual network interface assignment, inode assignment in the FS, CPU pinning and RAM allocation, and doing so does not violate our cloud-native principles. Whereas, PCI-PT doesn't just dedicate hardware resources, it exposes their underlying makeup, which then requires s/w custom written to communicate with the resources (and an exception in CNTT). In sriov-network-device-plugin, this custom s/w is instantiated in the host's kernel, which is consistent w/ CNTT architecture and principles. What is exposed to the container is an abstraction over a socket, so I'd suggest PCI-PT and sriov-network-device-plugin are intrinsically different (and if it's not, we should work with them to fix it).

If I'm missing something obvious, please help me with it.

Reply:
Let us remove it completely. i suggest. i.e. let us not have the exception in RA2, for now.

Reply:
@rabiabdel @CsatariGergely I'm fine w/ that, but this has implications for #1336

Reply:
I think whatever the conlusion in #1336 that should be concluded to the [Exception List of the RM](https://github.com/cntt-n/CNTT/blob/e13c1b9ce7e55df0fbd09a206bc5172c5e077bab/doc/ref_model/chapters/appendix-a.md#A.3) and this document should only refer to it.